battleShip.py

```diff
@@ -0,0 +1,278 @@
+                    continue
+                else:
+                    print("You missed!")
+                    self.mark_missed(x, y)
+
+                break
+            except ValueError:
+                print("Invalid coordinates. Try again :<")
+
+    def display_grid(self,opponent):  # Display the grid
+        size = len(grid)
+
+        # Display column labels
+        print("   ", end="")
+        for i in range(size):
+            print(f"{chr(i + ord('A')):2}", end=" ")
+        print()
+
+        # Display row labels and grid
+        for i in range(size):
+            print(f"{i+1:2}", end=" ")
+            for j in range(size):
+                if grid[i][j] == 'S '+ opponent.name: # if the owner is not the current player print -
+                    print("- ", end=" ")
+                elif grid[i][j] == 'S '+ self.name: # if the owner is the current player print S
+                    print("S ", end=" ")
+                else:
+                    print(grid[i][j], end=" ")
+            print()
+
+    def save_game(self, obj2):  # Save the game
+        with open("savegame.pickle", "wb") as file:
+            pickle.dump(obj2, file)
+            pickle.dump(self, file)
+            pickle.dump(grid, file) # Save the grid
+            
+        print("\t\tGame saved successfully :>)")
+       
+    def win_game(self,opponent):
+        if opponent.is_ship_destroyed():
+            print(f"\t\t\tKudos {self.name} wins! :>)")
+            return True
+        return False
+    
+    def turn_grid(self):
+        print(f"\n\t\t---- {self.name} Turn ----")
+        print(f"\t\t{self.name} Grid: \n")
+
+def display_menu():
+    print("\t\t----== Battle Ships Menu ----==")
+    print("\t\t1. New Game")
+    print("\t\t2. Load Game")
+    print("\t\t3. Quit")
+
+
+def display_menu2():
+    print("\t\t 1. Play with a friend")
+    print("\t\t 2. Play with a computer")
+    print("\t\t 3. Back to main menu")
+
+def load_game(): # Load the game
+    try:
+        list=[]
+        with open("savegame.pickle", "rb") as file:
+            global grid
+            p1=pickle.load(file)
+            list.append(p1)
+            p2=pickle.load(file)
+            list.append(p2)
+            grid=pickle.load(file)
+            return list
+    except FileNotFoundError:
+        print("No saved game found :<(")
+        return None
+    
+def check_win(player1,player2):
+    flag = False
+    while True:
+        player1.turn_grid()
+        player1.display_grid(player2)
+        player1.attack(player2)
+
+        if player1.win_game(player2):
+            flag =  True
+            return flag
+        
+        player2.turn_grid()
+        player2.display_grid(player1)
+        player2.attack(player1)
+
+        if player2.win_game(player1):
+            flag =  True
+            return flag
+
+def play_game(ships,ships_size,size,choice):
+    opponent = input("\t\tPlayer 1 Enter your name :) ")
```

Comment:
should ```play_game``` get the names of players or can it be extracted into another function :(